### ISSUE TITLE
read_imageh of image1d_array has wrong type of float coord parameter.

### ISSUE THREAD

**Issue by oddhack** [KhronosGroup/OpenCL-Registry#11]
Moved from internal issue 13235:

The extension specification for CL 2.0 has listed

```
half4 read_imageh(image1d_array_t image,
                  sampler_t sampler,
                  int2 coord);

half4 read_imageh(image1d_array_t image,
                  sampler_t sampler,
                  float4 coord);
```

Surely the float coordinates should be equivalent to all other read_imageX(image1d_array_t ...) and use float2, not float4?

In other words, the correct prototype should be:

```
half4 read_imageh(image1d_array_t image,
                  sampler_t sampler,
                  float2 coord);
```

Specification I'm looking at has: "Last Revision Date: 11/1/14"

Fixed in spec rev 29, but also needs changing in:
https://www.khronos.org/registry/cl/sdk/1.2/docs/man/xhtml/read_imageh1d.html

**Comment by Oblomov** [comment]
It seems that read_imagef has the same issue. The spec PDF for read_imageh (opencl-1.2-extensions) also says `float4`. But it really should be float2.

**Comment by oddhack** [comment]
Closed by #18
